### PR TITLE
Add memory discard related test cases

### PR DESCRIPTION
**Pull request by qijing-bj**
Signed-off-by: qijing <jinqi@redhat.com>

### CODE REVIEW

**Comment by qijing-bj** [comment]
# avocado run --vt-type libvirt libvirt_mem.positive_test.mem_basic
JOB ID     : b0ccec91b03afc899b4dbaef7c72065aaa5190b0
JOB LOG    : /root/avocado/job-results/job-2018-09-27T11.17-b0ccec9/job.log
 (1/3) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.mem_basic.default: PASS (119.33 s)
 (2/3) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.mem_basic.cold_plug_discard: PASS (114.50 s)
 (3/3) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.mem_basic.discard: PASS (137.87 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 375.84 s
# avocado run --vt-type libvirt libvirt_mem.positive_test.hugepages
JOB ID     : 8252882ffb2b8c985bf95828317951a3815a99f1
JOB LOG    : /root/avocado/job-results/job-2018-09-27T11.29-8252882/job.log
 (01/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.hot_plug: PASS (200.90 s)
 (02/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.hot_unplug: PASS (195.08 s)
 (03/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.cold_plug: PASS (172.20 s)
 (04/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.cold_plug_discard: PASS (179.33 s)
 (05/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.discard: PASS (183.68 s)
 (06/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.hot_plug: PASS (201.85 s)
 (07/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.hot_unplug: PASS (208.01 s)
 (08/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.cold_plug: PASS (172.31 s)
 (09/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.cold_plug_discard: PASS (170.06 s)
 (10/10) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.discard: PASS (203.78 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1892.80 s

File: libvirt/tests/cfg/libvirt_mem.cfg [hunk 1]
```diff
@@ -31,6 +31,14 @@
                     max_mem = 1536000
                     current_mem = 1536000
                     numa_cells = "{'id':'0','cpus':'0-1','memory':'1024000','unit':'KiB'} {'id':'1','cpus':'2-3','memory':'512000','unit':'KiB'}"
+                    variants:
+                        - default:
+                        - cold_plug_discard:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/libvirt_mem.cfg]: suggest change as below:
--discard:
            --variants:
                 --cold_plug
                 --hot_plug

Comment (jinqi7) [reply libvirt/tests/cfg/libvirt_mem.cfg]: No. 'default' is to keep the original test. 'discard' is to test discard with domain xml changed. 'cold_plug_discard' is to test memory device.

**Comment by dzhengfy** [comment]
@qijing-bj Please add meaningful commit messages and description in the PR. Thanks.

**Comment by chunfuwen** [comment]
@qijing-bj ,please merge 2 commits as 1

**Pull request merged**